nexus/types/src/external_api/params.rs

```diff
@@ -1427,21 +1483,59 @@ pub struct BgpPeerConfig {
 
     /// The address of the host to peer with.
     pub addr: IpAddr,
+
+    /// How long to hold peer connections between keppalives (seconds).
+    pub hold_time: u32,
+
+    /// How long to hold a peer in idle before attempting a new session
+    /// (seconds).
+    pub idle_hold_time: u32,
+
+    /// How long to delay sending an open request after establishing a TCP
+    /// session (seconds).
+    pub delay_open: u32,
+
+    /// How long to to wait between TCP connection retries (seconds).
+    pub connect_retry: u32,
+
+    /// How often to send keepalive requests (seconds).
+    pub keepalive: u32,
 }
 
 /// Parameters for creating a named set of BGP announcements.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
-pub struct CreateBgpAnnounceSet {
+pub struct BgpAnnounceSetCreate {
     #[serde(flatten)]
     pub identity: IdentityMetadataCreateParams,
 
     /// The announcements in this set.
-    pub announcement: Vec<BgpAnnouncement>,
+    pub announcement: Vec<BgpAnnouncementCreate>,
```

Comment:
Is this intrinsically bounded? We don't have many (any?) array inputs

Reply:
This is bound to the number of prefixes the rack uses for external-facing communications. I suspect in many deployments this will be a single entry. Some deployments will have a small handful due to the scarcity of IPv4 address space and having to fragment subnets. Going beyond double digits will be rare, I believe.